src/transaction.ts

```diff
@@ -191,28 +179,14 @@ export default class Transaction {
     if (includeSignature) {
       items = this.raw
     } else {
-      // EIP155 spec:
-      // If block.number >= 2,675,000 and v = CHAIN_ID * 2 + 35 or v = CHAIN_ID * 2 + 36, then when computing
-      // the hash of a transaction for purposes of signing or recovering, instead of hashing only the first six
-      // elements (i.e. nonce, gasprice, startgas, to, value, data), hash nine elements, with v replaced by
-      // CHAIN_ID, r = 0 and s = 0.
-
-      const v = bufferToInt(this.v)
-      const onEIP155BlockOrLater = this._common.gteHardfork('spuriousDragon')
-      const vAndChainIdMeetEIP155Conditions =
-        v === this._chainId * 2 + 35 || v === this._chainId * 2 + 36
-      const meetsAllEIP155Conditions = vAndChainIdMeetEIP155Conditions && onEIP155BlockOrLater
-
-      const unsigned = !this.r.length && !this.s.length
-      const seeksReplayProtection = this._chainId > 0
-
-      if ((unsigned && seeksReplayProtection) || (!unsigned && meetsAllEIP155Conditions)) {
```

Comment:
This is the heaviest code logic change in the PR and I get that this upper part has been moved into ``this._implementsEIP155()`` but I am not completely sure how to read this conditional simplification.

Could you give a short reasoning for the two cases in the ``if`` clause? That has likely already been done implicitly in your other explanations on the overall changes, but just want to make sure we are not missing anything here.

Won't make this a blocker request though.

Reply:
Actually, they aren't completely equivalent. Not for unsigned transactions.

The decision to use EIP155 or not when signing was previously based on the `_chainId` field, and its complex initialization logic in the constructor. It's simpler now: if the hardfork is spurious dragon or later, we use EIP155.

If a transaction is already signed and we are validating it, the logic is the same. It should be in spurious dragon or later and have an EIP155-encoded chain id as `v` to enable EIP155 validation.